src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -662,6 +662,19 @@ bool IdealLoopTree::is_range_check_if(IfNode *iff, PhaseIdealLoop *phase, Invari
   if (offset && !invar.is_invariant(offset)) { // offset must be invariant
     return false;
   }
+#ifndef PRODUCT
+  if (offset && phase->has_ctrl(offset)) {
+    Node* offset_ctrl = phase->get_ctrl(offset);
+    // If the control of offset is loop predication promoted by previous pass,
+    // then it will lead to cyclic dependency.
+    // Previously promoted loop predication is in the same loop of predication
+    // point.
+    if (phase->get_loop(predicate_proj) == phase->get_loop(offset_ctrl) &&
+        phase->is_dominator(predicate_proj, offset_ctrl)) {
+      assert(false, "willl lead to cyclic dependency");
```

Comment:
"willl" -> "will". Maybe you can also add "missed range check elimination" to the assertion message to make it more clear what the purpose of this assert is.